src/main/main.go

```diff
@@ -29,22 +30,44 @@ func main() {
 
 	log.Println("Plugin Version:", Version, "Build: ", Build)
 
-	// Fetch the registry cmd line options
-	flag.Var(&authorizedRegistries, "registry", "Specifies the authorized image registries")
-	flag.Parse()
+	// Fetch the registry from env
+	authorizedRegistries := os.Getenv("REGISTRY")
+
+	// Fetch the notary from env
+	authorizedNotary := os.Getenv("NOTARY")
+
+	// Fetch the notary RootCA from env
+	notaryRootCA := os.Getenv("NOTARY_ROOT_CA")
+
+	var notaryRootCAFolder = fmt.Sprintf("/root/.docker/tls/%s", authorizedNotary)
+	var notaryRootCAFile = fmt.Sprintf("%s/root-ca.crt", notaryRootCAFolder)
+	os.MkdirAll(notaryRootCAFolder, os.ModePerm)
+
+	f, err := os.Create(notaryRootCAFile)
+	errt := f.Truncate(0)
+  if (err != nil || errt != nil)  {
+    log.Fatal(err, errt)
+  }
+
+  defer f.Close()
+	_, err2 := f.WriteString(notaryRootCA)
+	if err2 != nil {
+    log.Fatal(err2)
+  }
 
 	// Convert authorized registries into a map for efficient lookup
 	registries := make(map[string]bool)
-	for _, registry := range authorizedRegistries {
```

Comment:
yes the README needs to be changed. It is not up to date with the current code. 

yes I agree with the map. We don't need it, but since it would require changes to other methods, I left it like that. As you wish. If you think we gain something from removing that map, then pls go ahead.

Reply:
Pushed minor change, keeping the registries map.